### PR TITLE
Herwig7: Finally arriving in CMSSW

### PR DESCRIPTION
**Pull request by mharrend**
Thanks to the help of @mkirsano Herwig7 can now be built and it works also together with the updated MG5aMC@NLO version.

### CODE REVIEW

**Comment by perrozzi** [comment]
+1

File: herwigpp.spec [hunk 1]
```diff
@@ -1,35 +1,67 @@
...
+Requires: openloops
+
+# Patch since otherwise Boost wants multithreaded lib, even though only single-threaded lib is installed
+# Problem exists since Herwig++3Beta
+
+Patch0: herwigpp-missingBoostMTLib
+
+
+BuildRequires: autotools
+BuildRequires: lhapdf
```
Comment (smuzaffar) [inline herwigpp.spec]: Cleanup this BuildRequire. lhapdf is already part of direct requires

**Comment by mharrend** [comment]
@smuzaffar lhapdf is needed both during the building of the Herwig7 rpm package and also later during the execution of Herwig7, so which of the two requirements can I safely drop?

**Comment by smuzaffar** [comment]
BuildRequires should be dropped.

**Comment by mharrend** [comment]
After taking @smuzaffar comment into account, can you please sign off the PR?

Thanks

**Comment by perrozzi** [comment]
+1

**Comment by smuzaffar** [comment]
cmsbuild, please test

File: herwigpp-toolfile.spec [hunk 2]
```diff
@@ -11,12 +11,16 @@ cat << \EOF_TOOLFILE >%i/etc/scram.d/herwigpp.xml
 <tool name="herwigpp" version="@TOOL_VERSION@">
   <client>
     <environment name="HERWIGPP_BASE" default="@TOOL_ROOT@"/>
-    <environment name="LIBDIR" default="$HERWIGPP_BASE/lib"/>
-    <environment name="INCLUDE" default="$HERWIGPP_BASE/include"/>
+    <environment name="LIBDIR" default="$HERWIGPP_BASE/lib/Herwig7"/>
+    <environment name="INCLUDE" default="$HERWIGPP_BASE/include/Herwig7"/>
   </client>
-  <runtime name="HERWIGPATH" value="$HERWIGPP_BASE/share/Herwig++"/>
+  <runtime name="HERWIGPATH" value="$HERWIGPP_BASE/share/Herwig7"/>
```
Comment (smuzaffar) [inline herwigpp-toolfile.spec]: @mharrend , please fix the toolfile to have correct path. There is no Herwig7 directory. I see only

```
slc6_amd64_gcc493/external/herwigpp/7.0.2/include/Herwig/ 
slc6_amd64_gcc493/external/herwigpp/7.0.2/lib/Herwig/  
slc6_amd64_gcc493/external/herwigpp/7.0.2/share/Herwig/
```

**Comment by smuzaffar** [comment]
@mharrend, can you fix the toolfile (to have correct paths) so that we can test this PR again?

**Comment by mharrend** [comment]
@smuzaffar I have updated the toolfile, please trigger again the tests. Thanks, Marco

**Comment by mharrend** [comment]
Can you please start the tests again? Thanks

**Comment by smuzaffar** [comment]
cmsbuild, please test

**Comment by smuzaffar** [comment]
@mharrend , can you please re-do this PR for IB/CMSSW_8_1_X/gcc530 (IB/CMSSW_8_1_X/stable is disabled and we do not have automatic tests enabled for stable branch). Please close this PR and create a new one for IB/CMSSW_8_1_X/gcc530

**Comment by mharrend** [comment]
Okay, I have created a new PR #2475 and will close this PR now.
BTW It is a little bit annoying that PR get not forwarded automatically to the proper branch if the old branch could not be used any longer.

**Comment by smuzaffar** [comment]
@mharrend , we could have created a PR for gcc530 branch ourself but then you would not have access to new PR (in case more changes were needed :-) )

**Comment by mharrend** [comment]
## Okay, fair enough. ;-)

Marco A. Harrendorf
KIT - IEKP

Am Mittwoch, den 31.08.2016, 01:33 -0700 schrieb Malik Shahzad
Muzaffar:

> @mharrend , we could have created a PR for gcc530 branch ourself but
> then you would not have access to new PR (in case more changes were
> needed :-) )

**Pull request closed without merging**